### PR TITLE
Ob 5362 upgrade to java17 and spring boot 3

### PR DESCRIPTION
**Pull request by tkuzynow**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

**Comment by vi-sudo** [comment]
Task linked: [OB-5362 Migrate agency service to Java 17](https://app.clickup.com/t/6656196/OB-5362)

File: src/main/java/de/caritas/cob/agencyservice/api/admin/service/agency/AgencyAdminSearchService.java [hunk 1]
```diff
@@ -73,140 +69,176 @@ public class AgencyAdminSearchService {
...
+        .ascending(
+            sort != null && sort.getOrder() != null ? sort.getOrder().equals(OrderEnum.ASC) : true)
+        .build();
+
+    try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+      queryResult = isBlank(keyword) || hasOnlySpecialCharacters(keyword)
+          ? searchAgenciesWithoutKeywordFilter(entityManager, agencyAdminSearch)
+          : searchAgenciesByKeyword(entityManager, agencyAdminSearch);
+    } catch (Exception ex) {
+      log.error("Count not create entity manager", ex);
```
Comment (SheepFromHeaven) [inline src/main/java/de/caritas/cob/agencyservice/api/admin/service/agency/AgencyAdminSearchService.java]:
```suggestion
      log.error("Could not create entity manager", ex);
```

File: src/test/java/de/caritas/cob/agencyservice/api/service/LogServiceTest.java [hunk 2]
```diff
@@ -34,7 +33,6 @@ public class LogServiceTest {
 
   @Before
   public void setup() {
```
Comment (SheepFromHeaven) [inline src/test/java/de/caritas/cob/agencyservice/api/service/LogServiceTest.java]: Can't we delete this?

**Review by SheepFromHeaven — DISMISSED** [review]
Approved with 2 small comments

File: src/main/java/de/caritas/cob/agencyservice/api/admin/service/agency/AgencyAdminSearchService.java (hunk 1 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/agencyservice/api/admin/service/agency/AgencyAdminSearchService.java]: fixed

File: src/test/java/de/caritas/cob/agencyservice/api/service/LogServiceTest.java (hunk 2 above)
Comment (tkuzynow) [reply src/test/java/de/caritas/cob/agencyservice/api/service/LogServiceTest.java]: fixed, thanks for spotting

**Review by web-mi — APPROVED** [review]

**Pull request merged**